src/common/roll_renderer.js

```diff
@@ -528,6 +529,18 @@ class Beyond20RollRenderer {
                 }
             }
 
+            for (let [dmgIndex, dmgType] of critical_damage_types.entries()) {
+                if (dmgType == "Colossus Slayer") {
+                    const dmg = critical_damages[dmgIndex].toString();
+                    if (dmg) {
+                        if (dmg_dice == "0") {
+                            critical_damages.splice(dmgIndex, 1);
+                            critical_damage_types.splice(dmgIndex, 1);
+                        }
+                    }
+                }
+            }
+
```

Comment:
The `if (dmg_dice =="0")` should be before the for loop, otherwise you're doing the for loop for nothing. Also, I don't know why you're not just doing the check/change inside the original `if (dmg_dice == "0")`, why have a separate loop for crits...
Also, I'm realizing.. why do we loop over the whole list to find Colossus Slayer instead of doing this : 
```js
const collossusSlayerIdx = damage_types.findIndex(d => d === "Colossus Slayer");
if (colossusSlayerIdx !== -1) {
  const dmg = damages[collossusSlayerIdx].toString();
  if (dmg) {
    const dmg_dice = await this.queryGeneric(request.name, `Add ${dmgType} damage ?`, { "0": "No", [dmg]: "Yes" }, "dmg_dice", ["0", dmg]);
   if (dmg_dice == "0") {
      const critIdx = critical_damage_types.findIndex(d => d === "Colossus Slayer");
      damages.splice(dmgIndex, 1);
      damage_types.splice(dmgIndex, 1);
      if (critIdx !== -1) {
        damages.splice(dmgIndex, 1);
        damage_types.splice(dmgIndex, 1);
     }
    }
  }
}